common/src/cformat.rs

```diff
@@ -203,19 +202,47 @@ impl CFormatSpec {
         }
     }
 
+    fn fill_string_with_precision(&self, string: String, fill_char: char) -> String {
```

Comment:
I split this into a separate function. Previously, it was coupled to `fill_string` with a boolean flag. It's a bit subjective, but I find it easier to reason about them as two separate functions.